library/Faktory/Job.hs

```diff
@@ -79,60 +59,52 @@ perform options producer arg = do
   jobJid job <$ pushJob producer job
 
 applyOptions :: Namespace -> JobOptions -> Job arg -> IO (Job arg)
-applyOptions namespace (JobOptions patches) = go patches
- where
-  go [] job = pure job
-  go (set : sets) job = case set of
-    SetRetry n -> go sets $ job { jobRetry = Just n }
-    SetQueue q ->
-      go sets $ job { jobQueue = Just $ namespaceQueue namespace q }
-    SetJobtype jt -> go sets $ job { jobJobtype = jt }
-    SetAt time -> go sets $ job { jobAt = Just time }
-    SetIn diff -> do
-      now <- getCurrentTime
-      go sets $ job { jobAt = Just $ addUTCTime diff now }
-
-retry :: Int -> JobOptions
-retry n = JobOptions [SetRetry n]
-
--- | Equivalent to @'retry' (-1)@: no retries, and move to Dead on failure
-once :: JobOptions
-once = retry (-1)
-
-queue :: Queue -> JobOptions
-queue q = JobOptions [SetQueue q]
-
-jobtype :: String -> JobOptions
-jobtype jt = JobOptions [SetJobtype jt]
-
-at :: UTCTime -> JobOptions
-at t = JobOptions [SetAt t]
-
-in_ :: NominalDiffTime -> JobOptions
-in_ i = JobOptions [SetIn i]
+applyOptions namespace options job = do
+  enqueueOptions <-
+    setAtFromIn $ namespaceQueue namespace $ options <> jobOptions job
```

Comment:
This should be reversed to `jobOptions job <> options` in the case, we do want the `options` (coming from `perform`) to override the `jobOptions` (set in `newJob`).

The `newJob` + `applyOptions` pattern is making less sense now, but I want to keep it this way because `newJob` is part of the module's exports, and I want this to be a minor version bump. When we go for major, we could clean this up to be less error-prone, and move/change `newJob` as necessary.